### PR TITLE
DOC-777: Create stream-worker-sentiment.md

### PR DESCRIPTION
**Pull request by diana-macrometa**
This PR adds a topic that explains the sentiment function, why you'd want to use it, and provides examples.

### CODE REVIEW

**Review by StoyanTc — APPROVED** [review]

**Review by koshy-thomas — APPROVED** [review]

File: docs/cep/stream-worker-sentiment.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+The `socialMediaSentimentAnalysis` processes social media posts from the `SocialMediaPostsStream`, including timestamp and username. It calculates the sentiment score for each post using the `sentiment:getRate()` function. The query outputs the timestamp, username, post, and the calculated sentiment score to the `SentimentAnalysisStream`.
+
+### Example 3
+
+```sql
+CREATE STREAM CustomerReviewsStream (productId string, timestamp long, username string, review string);
+
+CREATE TABLE AverageSentimentTable (productId string PRIMARY KEY, avgSentiment double);
```
Comment (koshy-thomas) [inline docs/cep/stream-worker-sentiment.md]: You cannot have primary key keyword here, if you want to store something as primary key for collection you need to store it as _key explicitly (select productId as _key, ..) and define (_key string, ...) in table definition.

File: docs/cep/stream-worker-sentiment.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+@info(name = 'customerReviewsSentimentAnalysis')
+INSERT INTO SentimentAnalysisStream
+SELECT productId, timestamp, username, review, sentiment:getRate(review) AS sentimentScore
+FROM CustomerReviewsStream;
+
+@info(name = 'averageSentimentPerProduct')
+INSERT INTO AverageSentimentTable
+SELECT productId, avg(sentimentScore) AS avgSentiment
+FROM SentimentAnalysisStream WINDOW TUMBLING_LENGTH(5)
+GROUP BY productId;
```
Comment (koshy-thomas) [inline docs/cep/stream-worker-sentiment.md:85]: Still if you use productId as _key you will just be inserting one value here and subsequent product reviews for that productId won't be inserted once the the tumbling length window is done .. so you will also need an UPDATE statement as well :

So I think the whole Stream worker should look something like this:

`CREATE STREAM CustomerReviewsStream (productId string, timestamp long, username string, review string);

CREATE TABLE AverageSentimentTable (_key string , avgSentiment double);

CREATE SINK STREAM SentimentAnalysisStream (productId string, timestamp long, username string, review string, sentimentScore int);

@info(name = 'customerReviewsSentimentAnalysis')
INSERT INTO SentimentAnalysisStream
SELECT productId, timestamp, username, review, sentiment:getRate(review) AS sentimentScore
FROM CustomerReviewsStream;

@info(name = 'averageSentimentPerProduct')
INSERT INTO AverageSentimentTable
SELECT productId as _key, avg(sentimentScore) AS avgSentiment
FROM SentimentAnalysisStream WINDOW TUMBLING_LENGTH(5)
GROUP BY productId;

@info(name = 'averageSentimentPerProductUpdate')
UPDATE AverageSentimentTable
SET AverageSentimentTable._key = _key, AverageSentimentTable.avgSentiment = avgSentiment
ON AverageSentimentTable._key == _key
SELECT productId as _key, avg(sentimentScore) AS avgSentiment
FROM SentimentAnalysisStream WINDOW TUMBLING_LENGTH(5)
GROUP BY productId;`

**Review by koshy-thomas — CHANGES_REQUESTED** [review]
refer to the above comments

**Comment by diana-macrometa** [comment]
Content has been updated, please re-review.

**Review by koshy-thomas — APPROVED** [review]
Looks good!

**Pull request merged**